go/vt/vtgate/executor.go

```diff
@@ -818,6 +825,49 @@ func (e *Executor) StreamExecute(ctx context.Context, method string, safeSession
 	return err
 }
 
+// handleMessageStream executes queries of the form 'stream * from t'
+func (e *Executor) handleMessageStream(ctx context.Context, safeSession *SafeSession, sql string, target querypb.Target, callback func(*sqltypes.Result) error, vcursor *vcursorImpl, logStats *LogStats) error {
+	stmt, err := sqlparser.Parse(sql)
+	if err != nil {
+		logStats.Error = err
+		return err
+	}
+
+	streamStmt, ok := stmt.(*sqlparser.Stream)
+	if !ok {
+		logStats.Error = err
+		return errors.New("preview identified a non-stream query as a stream query")
```

Comment:
This should be an internal error like here: https://github.com/youtube/vitess/blob/master/go/vt/vtgate/executor.go#L517.

Reply:
Good catch